### ISSUE TITLE
generics support

### ISSUE THREAD

**Issue by OneOfOne** [golang/vscode-go#1265]
Now that dev.typeparams have been merged into master and you can somewhat compile generic code, shouldn't the addon support the new syntax?

Not sure if this a vscode-go request or gopls (or both).

**Comment by findleyr** [comment]
I'm in the process of eliminating the API changes in Go 1.17 (see for example https://golang.org/cl/295929).  In that CL, I'm hiding some things behind the `go1.18` build constraint. It's theoretically possible that we could start building vscode-go / gopls features behind that build constraint as well, but at this point the API is subject to change and I think it would be premature to invest this effort (also, even the `go1.18` build constraint strategy is not set in stone at the moment).

So, while I agree that we should start adding support early, it's probably too early to do so now.

@hyangah do you want to leave this issue open, to track this effort when it is timely?

**Comment by hyangah** [comment]
Thanks for filing an issue @OneOfOne and sharing the status updates @findleyr 

Currently I don't expect much work is necessary from this extension side to support generics. We will need
  - gopls offering language features using the new API
  - gopls offering the semantic tokens for type parameters
  - go commands to parse, build and run the code with type parameters
  - other tools (dlv, ...) support code with type parameters and the runtime
  
 Most of them are blocked by stable API and spec. 
 
 I will close this issue for now. Let's reevaluate this once the spec and API become stable enough.
